### PR TITLE
Set up database for users and connected to API functions

### PR DESCRIPTION
**Pull request by glkidd**
Terence and I set up the Accounts table to hold our users data. Currently it just takes in a username and password for each user. We connected the database functions to the API functions for the sign up form so when you enter in a username and password and then click 'Submit' it will add your info to the table. In order to test this out, the front-end server, back-end server, and database server all need to be running at the same time locally. To run the front-end server, be in the cloud-haven-react-app directory then run 'npm start.' In another terminal window, outside of the react directory, run 'node server.js' to run the back-end. Reference the comment below to see how to set up the database on your local computer and then run it.

### CODE REVIEW

**Comment by turrence** [comment]
I've made a document on the google drive to see how to setup your local dynamodb database. [here](https://docs.google.com/document/d/1pd8TmqJ4dSkbAucMj4C2GxsRmqLf6SHxbIlFxssOeWc/edit). I've setup up aliases for a lotta commands you see in the doc because we will be using these commands to run the database. 

**One thing to note about running the `aws` commands that refer to the local database is that they have to include `--endpoint-url http://localhost:8000`**

## for e.g 
```
aws dynamodb list-tables --endpoint-url http://localhost:8000
``` 

Going through the doc should tell you how to set it up locally

File: dao.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  endpoint: "http://localhost:8000"
+});
+
+var dynamo = function(req, res) {
+    this.tableName = "Accounts";
+    this.docClient = new AWS.DynamoDB.DocumentClient();
+}
+
+//var tableName = "Accounts";
+//var docClient = new AWS.DynamoDB.DocumentClient();
```
Comment (Michelle5102) [inline dao.js:14]:
```suggestion
```
I think these are extra comments accidentally left in

File: dao.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    this.docClient.get(params, function(err, data) {
+        if (err) {
+            console.error("Unable to read item. Error JSON:", JSON.stringify(err, null, 2));
+        } else {
+            console.log("GetItem succeeded:", JSON.stringify(data, null, 2));
+        }
+    });
+}
+
+module.exports = dynamo;
```
Comment (Michelle5102) [inline dao.js:51]:
```suggestion
module.exports = dynamo;

```
Newline needed for GitHub to diff files properly

File: dynamoDb.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+aws dynamodb create-table \
+    --table-name Accounts \
+    --attribute-definitions \
+        AttributeName=Username,AttributeType=S \
+    --key-schema \
+        AttributeName=Username,KeyType=HASH \
+    --provisioned-throughput \
+        ReadCapacityUnits=10,WriteCapacityUnits=5 \
+    --endpoint-url http://localhost:8000
```
Comment (Michelle5102) [inline dynamoDb.sh:14]:
```suggestion
    --endpoint-url http://localhost:8000

```
Newline

File: dao.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+//var tableName = "Accounts";
+//var docClient = new AWS.DynamoDB.DocumentClient();
+
+dynamo.prototype.putUser = function(username, password) {
+    var params = {
+        TableName: this.tableName,
+        Item : {
+            "Username" : username,
+            "Password" : password
```
Comment (Michelle5102) [inline dao.js:21]:
```suggestion
        tableName: this.tableName,
        item : {
            "username" : username,
            "password" : password
```
Our style guide says: "parameters and local variables are all lowerCamelCase." Is this something specifically needed for setting up the database? I think these are like parameters, but now I'm a bit confused if the upper case letter is for something specific

File: dao.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+            console.log("Added item:", JSON.stringify(data, null, 2));
+        }
+    });
+}
+
+dynamo.prototype.getUser = function(username) {
+    var params = {
+        TableName: this.tableName,
+        Key : {
+            "Username" : username
```
Comment (Michelle5102) [inline dao.js:38]:
```suggestion
        tableName: this.tableName,
        key : {
            "username" : username
```

File: cloud-haven-react-app/src/components/SignUp/SignUp.js [hunk 6]
```diff
@@ -19,7 +20,8 @@ const SignUp = () => {
             "email": email,
             "password": password
         };
-        alert(JSON.stringify(signUpData))
+        //alert(JSON.stringify(signUpData))
```
Comment (Michelle5102) [inline cloud-haven-react-app/src/components/SignUp/SignUp.js:23]:
```suggestion
```

**Review by patrickbartlett — APPROVED** [review]

**Review by Michelle5102 — CHANGES_REQUESTED** [review]
Overall it looks good! A couple newlines needed for GitHub and style things. Are some of the parameters upper camal case for a database related reason?

File: dao.js (hunk 5 above)
Comment (Michelle5102) [reply dao.js:36]: Sounds like the upper camel case is needed for the database.

File: dao.js (hunk 4 above)
Comment (Michelle5102) [reply dao.js:19]: Sounds like the upper camel case is needed for the database.

File: dynamoDb.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (Michelle5102) [inline dynamoDb.sh:15]:
```suggestion

```
There are some spaces here. I must've accidentally left them in my last suggested change. Sorry!

**Review by Michelle5102 — APPROVED** [review]
Looks good!

**Pull request merged**